src/stylesheets/uswds.scss

```diff
@@ -15,3 +15,10 @@
 // Utilities
 // -------------------------------------
 @import "packages/uswds-utilities";
+
+
+// Test code to fix dropdown error text issue.
+// REMOVE BEFORE MERGE
+#select-error-fix {
+    padding: .25rem 0 0 .5rem;
```

Comment:
Let's try to avoid magic numbers like this. If we update the border-width or the component padding we'll have to update this value again.

In breadcrumb we calculated some of the paddings with `calc()`. You can see an example below:
https://github.com/uswds/uswds/blob/514776640ab303fd2ff4bbdbdf3ed524568e9a7b/src/stylesheets/components/_breadcrumb.scss#L41

We could do something similar with text-input in [`.usa-input--error`](https://github.com/uswds/uswds/blob/514776640ab303fd2ff4bbdbdf3ed524568e9a7b/src/stylesheets/elements/form-controls/_text-input.scss#L11-L13):

```scss
.usa-input--error {
  @include u-border($theme-input-state-border-width, "error-dark");
  padding: calc(#{units(1)} - #{units($theme-input-state-border-width)});
}
```

`units(1)` comes from the padding set in `%block-input-styles`. Thoughts @thisisdano?

Reply:
Added @mejiaj calc solution and tested it on multiple text input components. All seem to work except for combo box and time picker 
![image](https://user-images.githubusercontent.com/25211650/123853173-25eaef80-d8eb-11eb-9427-730d42db25eb.png)

The error border may be useful for the character limit component!
![image](https://user-images.githubusercontent.com/25211650/123853409-6ba7b800-d8eb-11eb-909d-13622e02e884.png)

Reply:
You'd have to pass the `usa-input--error` class to the inputs themselves with Javascript. But once applied they look as expected.

![image](https://user-images.githubusercontent.com/3385219/124641234-153efa00-de54-11eb-9f46-75051f20d716.png)